test/integ/pandas/test_pandas_tools.py

```diff
@@ -300,6 +300,21 @@ def test_invalid_table_type_write_pandas(
             )
 
 
+def test_empty_dataframe_write_pandas(
+    conn_cnx: Callable[..., Generator[SnowflakeConnection, None, None]],
+):
+    table_name = random_string(5, "empty_dataframe_")
+    df = pandas.DataFrame([], columns=["name", "balance"])
+    with conn_cnx() as cnx:
+        try:
+            success, num_chunks, num_rows, _ = write_pandas(
+                cnx, df, table_name, auto_create_table=True
+            )
+            assert success and num_chunks == 1 and num_rows == 0
+        finally:
+            cnx.execute_string(f"DROP TABLE IF EXISTS {table_name}")
```

Comment:
No need to call that helper fn (execute is enough), also if we used a temp table then this wouldn't be necessary at all.
```suggestion
            cnx.execute(f"DROP TABLE IF EXISTS {table_name}")
```